packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -88,6 +93,25 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
     static Texture: Constructor<Texture>;
     static Image: Constructor<Image>;
 
+    async createTexture(uri: string): Promise<ModelViewerTexture|null> {
+      const currentGLTF = this[$currentGLTF];
+      if (currentGLTF) {
+        const texture = await new TextureLoader().load(uri);
+        const textures = new Set<Texture>();
+        const textureDef = new GLTFTextureDefinition();
+        const imageDef = new GLTFImageDefinition(uri);
+        textures.add(texture);
+
+        const {gltf} = currentGLTF.correlatedSceneGraph;
+
+        return new ModelViewerTexture(
+            () => {}, gltf, textureDef, textures, imageDef);
```

Comment:
I think we need our `onUpdate()` function here or else if any setting is changed on a descendent of this texture (say a sampler) it won't cause our render to redraw.

Reply:
Done.